From e3683b2992a74c03dcbcd73fba690986f121ea88 Mon Sep 17 00:00:00 2001
From: Giuseppe Scrivano <gscrivan@redhat.com>
Date: Thu, 3 Oct 2013 14:48:09 +0200
Subject: [PATCH 1/2] virtinst: expose disk/source startupPolicy attribute

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>
(cherry picked from commit f5a7476497ee733458da79294ba06860ad82d8c7)

Conflicts:
	virtinst/devicedisk.py
---
 virtinst/VirtualDisk.py | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/virtinst/VirtualDisk.py b/virtinst/VirtualDisk.py
index 54f072d..8ad9cae 100644
--- a/virtinst/VirtualDisk.py
+++ b/virtinst/VirtualDisk.py
@@ -540,7 +540,7 @@ class VirtualDisk(VirtualDevice):
                  volInstall=None, volName=None, bus=None, shareable=False,
                  driverCache=None, selinuxLabel=None, format=None,
                  validate=True, parsexml=None, parsexmlnode=None, caps=None,
-                 driverIO=None, sizebytes=None):
+                 driverIO=None, sizebytes=None, startupPolicy=None):
         """
         @param path: filesystem path to the disk image.
         @type path: C{str}
@@ -586,6 +586,9 @@ class VirtualDisk(VirtualDevice):
         @param sizebytes: Optionally specify storage size in bytes. Takes
                           precedence over size if specified.
         @type sizebytes: C{int}
+        @param startupPolicy: Define policy what to do with the disk if
+                              the source file is not accessible.
+        @type startupPolicy: C{str}
         """
 
         VirtualDevice.__init__(self, conn=conn,
@@ -613,6 +616,7 @@ class VirtualDisk(VirtualDevice):
         self._error_policy = None
         self._serial = None
         self._target = None
+        self._startup_policy = None
         self._validate = validate
 
         # XXX: No property methods for these
@@ -642,6 +646,7 @@ class VirtualDisk(VirtualDevice):
         self._set_selinux_label(selinuxLabel, validate=False)
         self._set_format(format, validate=False)
         self._set_driver_io(driverIO, validate=False)
+        self._set_startup_policy(startupPolicy, validate=False)
 
         self.__change_storage(self.path,
                               self.vol_object,
@@ -888,6 +893,14 @@ class VirtualDisk(VirtualDevice):
     serial = _xml_property(_get_serial, _set_serial,
                            xpath="./serial")
 
+
+    def _get_startup_policy(self):
+        return self._startup_policy
+    def _set_startup_policy(self, val, validate=True):
+        self._startup_policy = val
+    startup_policy = _xml_property(_get_startup_policy, _set_startup_policy,
+                                   xpath="./source/@startupPolicy")
+
     # If there is no selinux support on the libvirt connection or the
     # system, we won't throw errors if this is set, just silently ignore.
     def _get_selinux_label(self):
@@ -1500,7 +1513,10 @@ class VirtualDisk(VirtualDevice):
                 ret += "      <driver%s/>\n" % drvxml
 
         if path is not None:
-            ret += "      <source %s='%s'/>\n" % (typeattr, path)
+            ret += "      <source %s='%s'" % (typeattr, path)
+            if self.startup_policy:
+                ret += " startupPolicy='%s'" % self.startup_policy
+            ret += "/>\n"
 
         bus_xml = ""
         if self.bus is not None:
-- 
1.9.0

